storage/client/src/main/java/com/azure/storage/blob/AppendBlobClientBuilder.java

```diff
@@ -106,28 +126,76 @@ public AppendBlobAsyncClient buildAsyncClient() {
     }
 
     /**
-     * Sets the service endpoint, additionally parses it for information (SAS token, container name)
+     * Sets the service endpoint, additionally parses it for information (SAS token, container name, blob name)
      * @param endpoint URL of the service
      * @return the updated AppendBlobClientBuilder object
      */
     public AppendBlobClientBuilder endpoint(String endpoint) {
         Objects.requireNonNull(endpoint);
+        URL url;
         try {
-            this.endpoint = new URL(endpoint);
+            url = new URL(endpoint);
+            this.endpoint = new URL(url.getProtocol() + "://" + url.getAuthority());
+            String path = url.getPath();
+            if (path != null && !path.isEmpty() && !path.equals("/")) {
+                path = path.replaceAll("^/", "").replaceAll("/$", "");
+                String[] segments = path.split("/", 2);
+                if (segments.length != 2) {
+                    throw new IllegalArgumentException("Endpoint should contain 0 or at least 2 path segments");
```

Comment:
This is not always true. If the customer has set up a root container, it's possible to have a blobUrl of the format:
myaccount.blob.core.windows.net/blobName

Reply:
I'm using BlobUrlParts now with UrlParser.